paasta_tools/cli/cmds/secret.py

```diff
@@ -482,13 +481,8 @@ def paasta_secret(args):
             # make it clear that we're running in a sub-shell
             new_environ["PS1"] = r"(paasta secret run) \$ "
 
-        command_to_run = subprocess.run(
-            args.cmd,
-            shell=False,
-            check=False,
-            env=new_environ,
-        )
-        sys.exit(command_to_run.returncode)
+        # Note: this never returns, it replaces the current process
```

Comment:
imo, adding the pgctl context here might be worth doing - otherwise, someone might try to "clean" this up in the future and break things :)